addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,317 @@
+        return super(HrExpense, self)._track_subtype(init_values)
+
+    def _line_get_convert(self, x, part, date):
+        partner_id = self.env['res.partner']._find_accounting_partner(part).id
+        return {
+            'date_maturity': x.get('date_maturity'),
+            'partner_id': partner_id,
+            'name': x['name'][:64],
+            'date': date,
+            'debit': x['price']>0 and x['price'],
+            'credit': x['price']<0 and -x['price'],
+            'account_id': x['account_id'],
+            'analytic_line_ids': x.get('analytic_line_ids'),
+            'amount_currency': x['price']>0 and abs(x.get('amount_currency')) or -abs(x.get('amount_currency')),
+            'currency_id': x.get('currency_id'),
+            'tax_line_id': x.get('tax_line_id'),
+            'ref': x.get('ref'),
+            'quantity': x.get('quantity',1.00),
+            'product_id': x.get('product_id'),
+            'product_uom_id': x.get('uos_id'),
+            'analytic_account_id': x.get('account_analytic_id'),
+        }
+
+    @api.multi
+    def _compute_expense_totals(self, company_currency, account_move_lines):
+        '''
+        internal method used for computation of total amount of an expense in the company currency and
+        in the expense currency, given the account_move_lines that will be created. It also do some small
+        transformations at these account_move_lines (for multi-currency purposes)
+        
+        :param account_move_lines: list of dict
+        :rtype: tuple of 3 elements (a, b ,c)
+            a: total in company currency
+            b: total in hr.expense currency
+            c: account_move_lines potentially modified
+        '''
+        self.ensure_one()
+        total = 0.0
+        total_currency = 0.0
+        for line in account_move_lines:
+            line['currency_id'] = False
+            line['amount_currency'] = False
+            if self.currency_id != company_currency:
+                line['currency_id'] = self.currency_id.id
+                line['amount_currency'] = line['price']
+                line['price'] = self.currency_id.with_context(date=self.date or fields.Date.context_today(self)).compute(line['price'], company_currency)
+            total -= line['price']
+            total_currency -= line['amount_currency'] or line['price']
+        return total, total_currency, account_move_lines
+
+    @api.multi
+    def action_move_create(self):
+        '''
+        main function that is called when trying to create the accounting entries related to an expense
+        '''
+        if any(expense.state != 'approve' for expense in self):
+            raise UserError(_("You can only generate accounting entry for approved expense(s)."))
+
+        company_id = self.env['res.company']._company_default_get('hr.employee')
+        journal_id = self.env['account.journal'].search([('type', '=', 'purchase'), ('company_id', '=', company_id)], limit=1).id
```

Comment:
I think, no need to pass company domain here.

we already have multi company record rules on account.journal.
see: https://github.com/odoo/odoo/blob/master/addons/account/security/account_security.xml#L50